From 8ef0c1a24ebc8dc52fcef6fe81167ef2942c1e36 Mon Sep 17 00:00:00 2001
From: Alexandre Ghiti <alexandre.ghiti@canonical.com>
Date: Tue, 2 Nov 2021 18:18:10 +0100
Subject: [PATCH 5/5] riscv: Add target to create SPL binary for BeagleV

The ZSBL (ddrinit) needs the next boot stage firmware to have a header
that contains the size of the firmware.

The command to do so comes from Buildroot.

Signed-off-by: Alexandre Ghiti <alexandre.ghiti@canonical.com>
---
 arch/riscv/config.mk | 10 ++++++++++
 1 file changed, 10 insertions(+)

--- a/arch/riscv/config.mk
+++ b/arch/riscv/config.mk
@@ -32,3 +32,13 @@
 
 EFI_CRT0		:= crt0_riscv_efi.o
 EFI_RELOC		:= reloc_riscv_efi.o
+
+ifdef CONFIG_TARGET_STARFIVE_JH7100
+quiet_cmd_spl_dtb_bin_out = BIN_OUT $@
+      cmd_spl_dtb_bin_out = perl -e 'print pack("l", (stat @ARGV[0])[7])' $< > $@ && cat $< >> $@
+
+$(obj)/u-boot-spl-dtb.bin.out: $(obj)/u-boot-spl-dtb.bin
+	$(call cmd,spl_dtb_bin_out)
+
+INPUTS-$(CONFIG_SPL_BUILD) += $(obj)/u-boot-spl-dtb.bin.out
+endif
